consensus/polybft/stake_manager_fuzz_test.go

```diff
@@ -0,0 +1,205 @@
+package polybft
+
+import (
+	"encoding/json"
+	"math/big"
+	"testing"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	go_fuzz_utils "github.com/trailofbits/go-fuzz-utils"
+)
+
+func FuzzTestStakeManagerPostEpoch(f *testing.F) {
+	state := newTestState(f)
+
+	seeds := []struct {
+		EpochID    uint64
+		Validators AccountSet
+	}{
+		{
+			EpochID:    0,
+			Validators: newTestValidators(f, 6).getPublicIdentities(),
+		},
+		{
+			EpochID:    1,
+			Validators: newTestValidators(f, 42).getPublicIdentities(),
+		},
+		{
+			EpochID:    42,
+			Validators: newTestValidators(f, 6).getPublicIdentities(),
+		},
+	}
+
+	for _, seed := range seeds {
+		data, _ := json.Marshal(seed)
+		f.Add(data)
+	}
+
+	f.Fuzz(func(t *testing.T, input []byte) {
+		stakeManager := &stakeManager{
+			logger:              hclog.NewNullLogger(),
+			state:               state,
+			maxValidatorSetSize: 10,
+		}
+
+		tp, err := go_fuzz_utils.NewTypeProvider(input)
+		if err != nil {
+			return
+		}
+		newEpochID, err := tp.GetUint64()
+		if err != nil {
+			return
+		}
+		var validators AccountSet
+		err = tp.Fill(validators)
+		if err != nil {
+			return
+		}
+
+		_ = stakeManager.PostEpoch(&PostEpochRequest{
```

Comment:
Consider adding error check here. WDYT?

Reply:
We can not assert any function responses because we don't know what will fuzzer generate. We can only compare responses (if we had that case, e.g. two functions that should have same response)